bika/lims/content/attachment.py

```diff
@@ -177,33 +185,30 @@ def getAnalysis(self):
         """Return the analysis to which this is linked it may not be linked to
         an analysis
         """
-        tool = getToolByName(self, REFERENCE_CATALOG)
-        uids = [uid for uid in
-                tool.getBackReferences(self, 'AnalysisAttachment')]
-        if len(uids) == 1:
-            reference = uids[0]
-            analysis = tool.lookupObject(reference.sourceUID)
-            return analysis
-        return None
+        analysis = get_backreferences(self, 'AnalysisAttachment',
+                                      as_brains=True)
+        if not analysis:
+            return None
+
+        if len(analysis) > 1:
+            logger.warn("Single attachment assigned to more than one Analysis")
+
+        analysis = api.get_object(analysis[0])
+        return analysis
 
     def getParentState(self):
         """Return the review state of the object - analysis or AR to which
         this is linked
         """
-        tool = getToolByName(self, REFERENCE_CATALOG)
-        uids = [uid for uid in
-                tool.getBackReferences(self, 'AnalysisAttachment')]
-        if len(uids) == 1:
-            reference = uids[0]
-            parent = tool.lookupObject(reference.sourceUID)
-        else:
-            uids = [uid for uid in
-                    tool.getBackReferences(self, 'AnalysisRequestAttachment')]
-            if len(uids) == 1:
-                reference = uids[0]
-                parent = tool.lookupObject(reference.sourceUID)
-        workflow = getToolByName(self, 'portal_workflow')
-        return workflow.getInfoFor(parent, 'review_state', '')
+        analysis = self.getAnalysis()
+        if analysis:
+            return getCurrentState(analysis)
+
+        analysis_request = self.getRequest()
+        if analysis_request:
+            return getCurrentState(analysis_request)
+
+        return ''
```

Comment:
Not sure about the possibilities to come to this point in code. Shall we add a  `logger.warn('Free floating attachment detected: {}'.format(repr(self)))`?

Reply:
I've removed `getParentState` function. Is not used anywhere in the code and seems quite awkward to me